web/src/components/product/ProductPage.jsx

```diff
@@ -0,0 +1,435 @@
+ * @param {boolean} props.isDisabled
+ */
+const DeregisteredButtons = ({ isDisabled = false }) => {
+  const [isRegistrationPopupOpen, setIsRegistrationPopupOpen] = useState(false);
+  const [isSelectionPopupOpen, setIsSelectionPopupOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        variant="primary"
+        onClick={() => setIsRegistrationPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Register")}
+      </Button>
+      <Button
+        variant="secondary"
+        onClick={() => setIsSelectionPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Change product")}
+      </Button>
+      <ProductRegistrationPopup
+        isOpen={isRegistrationPopupOpen}
+        onFinish={() => setIsRegistrationPopupOpen(false)}
+        onCancel={() => setIsRegistrationPopupOpen(false)}
+      />
+      <ProductSelectionPopup
+        isOpen={isSelectionPopupOpen}
+        onFinish={() => setIsSelectionPopupOpen(false)}
+        onCancel={() => setIsSelectionPopupOpen(false)}
+      />
+    </>
+  );
+};
+
+/**
+ * Buttons for a product that is already registered.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isDisabled
+ */
+const RegisteredButtons = ({ isDisabled = false }) => {
+  const [isDeregistrationPopupOpen, setIsDeregistrationPopupOpen] = useState(false);
+  const [isWarningPopupOpen, setIsWarningPopupOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        variant="primary"
+        onClick={() => setIsDeregistrationPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Deregister")}
+      </Button>
+      <Button
+        variant="secondary"
+        onClick={() => setIsWarningPopupOpen(true)}
+        isDisabled={isDisabled}
+      >
+        {_("Change product")}
+      </Button>
+      <ProductDeregistrationPopup
+        isOpen={isDeregistrationPopupOpen}
+        onFinish={() => setIsDeregistrationPopupOpen(false)}
+        onCancel={() => setIsDeregistrationPopupOpen(false)}
+      />
+      <RegisteredWarningPopup
+        isOpen={isWarningPopupOpen}
+        onAccept={() => setIsWarningPopupOpen(false)}
+      />
+    </>
+  );
+};
+
+/**
+ * Renders the actions for the current product.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isDisabled
+ */
+const ProductActions = ({ isDisabled = false }) => {
+  const { registration } = useProduct();
+
+  const withRegistration = registration.requirement !== "not-required";
+  const registered = registration.code !== null;
+
+  return (
+    <>
+      <div className="split">
+        <If
+          condition={withRegistration}
+          then={
+            <If
+              condition={registered}
+              then={<RegisteredButtons isDisabled={isDisabled} />}
+              else={<DeregisteredButtons isDisabled={isDisabled} />}
+            />
+          }
+          else={<WithoutRegistrationButtons isDisabled={isDisabled} />}
+        />
```

Comment:
Wow! You manage to find a place where use a nested <If> :stuck_out_tongue_winking_eye: